source/common/stats/isolated_store_impl.h

```diff
@@ -113,6 +113,11 @@ class IsolatedStoreImpl : public StoreImpl {
     Histogram& histogram = histograms_.get(name, unit);
     return histogram;
   }
+  Histogram& histogramFromStatName(StatName name, const std::vector<Tag>&,
+                                   Histogram::Unit unit) override {
+    Histogram& histogram = histograms_.get(name, unit);
```

Comment:
this is obviously not right, might need to key the metrics off a pair of name + tag?

Reply:
I think the issue is that so far is that just using the original name of the histogram is sufficient to uniquely identify it: since the tag extracted name + tags is a function of original metric name + bootstrap config, tags can be ignored when looking up cached metric objects. It seems like the stat stores are using the full stat name as the hash key.

I'll play around with some ways of encoding the static tags into the hash key

Reply:
I think the easiest mechanism would be to synthesize a StatName the incorporates the tag names & values, e.g.

   scope.subsystem.counter1 [tag1=value1, tag2=value2]
could be encoded to have a StatName of:
   scope.subsystem.counter1.TAGS.tag1.value1.tag2.value2
this would be a memory-efficient encoding because the tags and values would all be references to the same symbols.

Then you don't have to change the hashing at all, and they can be stored in the same table as the existing stats.

Reply:
Side comment: it would be cool if we could preserve the positions of the tags in the synthesized string. That way we can optimize away 1) the regex to extract tags back for prometheus, 2) linear number of regexes for each tag name.